app/recipe/api.py

```diff
@@ -0,0 +1,49 @@
+
+from .dto import RecipeDto, IngredientDto
+from .services import RecipeService
+
+
+class RecipeApi:
+
+    @staticmethod
+    def get_all_recipes():
+        recipes = RecipeService.get_all_recipes()
+        recipes_dtos = []
+
+        for recipe in recipes:
+            ingredients = RecipeService.get_ingredients(recipe.pk)
```

Comment:
Not an issue here but it's often not the best practice to get things from the database inside loops (basically since we have to do a database call for every recipe individually to get the ingredients). No need to change or anything, just an fyi